clients/ctl/admin-ui/cypress/components/ClassifiedDataCategoryInput.cy.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from "react";
+
+import ClassifiedDataCategoryDropdown from "~/features/dataset/ClassifiedDataCategoryDropdown";
+import { MOCK_DATA_CATEGORIES } from "~/mocks/data";
+import { DataCategory } from "~/types/api";
+
+const CATEGORIES_WITH_CONFIDENCE = MOCK_DATA_CATEGORIES.map((dc) => ({
+  ...dc,
+  confidence: Math.floor(Math.random() * 100),
+}));
+const MOST_LIKELY_CATEGORIES = CATEGORIES_WITH_CONFIDENCE.slice(0, 5);
+
+describe("ClassifiedDataCategoryDropdown", () => {
+  it("can render checked categories", () => {
+    const onCheckedSpy = cy.spy().as("onCheckedSpy");
+    const selectedClassifiedCategory = MOST_LIKELY_CATEGORIES[0];
+    cy.mount(
+      <ClassifiedDataCategoryDropdown
+        dataCategories={MOCK_DATA_CATEGORIES as DataCategory[]}
```

Comment:
If the definition of `MOCK_DATA_CATEGORIES` had the api type, or used `as const`, these casts wouldn't be necessary.

Reply:
oh right! I meant to talk about this with you. So the openapi type has type `parent_key?: string`. however, it turns out `parent_key = null` in real life! I'm not sure if this is just the typescript generator not being good about null vs undefined or something about our backend definition, but that's why I had this cast

Reply:
Still digging into the OpenApi schema, but the answer may be that we should use [response_model_exclude_none](https://fastapi.tiangolo.com/tutorial/response-model/?h=response_model_exclude_none#use-the-response_model_exclude_unset-parameter) on our routes to exclude optionals instead of passing nulls. Applying that to all crud endpoints would be very nice for consistency.

As-is, I think the TS generator is _technically_ correct because if you look directly at the OpenAPI schema it says the value is optional (not required) and does _not_ say the type can be `null`:
<img width="492" alt="Screen Shot 2022-09-30 at 16 59 19" src="https://user-images.githubusercontent.com/2236777/193370633-32622a86-fe4a-444a-a3d9-b902937371fc.png">
(`*` by a field indicates it is required)

I don't know of a way to get fastapi/pydantic to say the type can be `null` -- [this](https://fastapi.tiangolo.com/it/tutorial/query-params-str-validations/#required-with-none) didn't do it when I tried -- and I would rather exclude them anyway.